### ISSUE TITLE
[Add Test] T:System.Xml.Serialization.SoapReflectionImporter

### ISSUE THREAD

**Issue by shmao** [dotnet/corefx#10579]
T:System.Xml.Serialization.SoapAttributeAttribute
T:System.Xml.Serialization.SoapAttributeOverrides
T:System.Xml.Serialization.SoapAttributes
T:System.Xml.Serialization.SoapElementAttribute
T:System.Xml.Serialization.SoapEnumAttribute
T:System.Xml.Serialization.SoapIgnoreAttribute
T:System.Xml.Serialization.SoapIncludeAttribute
T:System.Xml.Serialization.SoapReflectionImporter
T:System.Xml.Serialization.SoapTypeAttribute

**Comment by shmao** [comment]
SoapReflectionImporter is for generating mappings for .NET Framework types which require SOAP-encoded messages. And the mappings are used by XmlSerializer to generate serializers for those types. The full framework implementation of XmlSerializer uses CodeDOM to generate dynamic serializer assembly when any type require Soap encoding. Because CodeDOM is not available on CoreCLR and we don't have any plans to support generating code to execute at runtime, serializing types into soap encoded message wouldn't work. The issues is being blocked by CodeDOM missing on CoreCLR. We would postpone adding these APIs.

**Comment by Petermarcu** [comment]
Is this in netstandard2.0 per the label? If so, please put it in the 1.2 milestone.

**Comment by danmosemsft** [comment]
@zhenlan @shmao codeDOM is now present but only for code generation. If we cannot add a proper implementation of these API's then we should please have an issue in the Standard repo to consider removing the API's (unless we plan to throw from them)

**Comment by shmao** [comment]
@danmosemsft We had a discussion on this. To support these APIs, without CodeDom, we potentially can implement a new solution either using RefEmit or reflection based serialization. We expected it to be at least 1 dev month work. The decision for the APIs was to leave them in standard and we prioritize the work for Core as necessary based on the usage.

/cc: @weshaggard

**Comment by danmosemsft** [comment]
We should track that with an issue in the Standard repo.

**Comment by weshaggard** [comment]
I think we should keep the APIs and stub/throw for them until we can properly implement them.

**Comment by zhenlan** [comment]
> I think we should keep the APIs and stub/throw for them until we can properly implement them

I normally don't like stub/throw because it makes debugging even harder, but it is acceptable if we already do this for other APIs. Do we?

**Comment by zhenlan** [comment]
After some offline discussion, we decided to keep these APIs in NS2.0. Some functionality will not work on .NET Core without CodeDom or re-implementing the feature. We will deal with that as bug fix in the future, but keeping these APIs will allow apps to compile across different frameworks. For now, we do need to make sure corresponding functions will fail with clear message (eg. throws PNSE) instead of silent failure or alike.

**Comment by shmao** [comment]
Some tests for this had been added by PR #12909.

@huanwu can you please see if we have enough tests for System.Xml.Serialization.SoapReflectionImporter?

**Comment by huanwu** [comment]
Did the ad-hoc tests. The returned mapping object seems the same as the one get from desktop version.

**Comment by shmao** [comment]
Thanks for confirming this, @huanwu . Closing the issue.
